plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
+                                                                    y: [ //array of labels in bar chart yaxis ]
+                                                                    graphic_link_pattern: // array of links which are used to make chart clickable
+                                                                },
+                                                                fullData:{
+                                                                    // Data with default parameters used by plotly
+                                                                    alignmentgroup:"",
+                                                                    cliponxaxis: true,
+                                                                    x:[],
+                                                                    y:[]
+                                                                }
+                                                                label: //label on which click event took place
+                                                                text:""
+                                                            }
+                                                        }
+                                                    }
+                                                    
+                                                    
+                                                    */
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx="";
+                                                        switch(data.points[0].data.type){
+                                                            case "pie":
+                                                                idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
+                                                                break;
+                                                            case "bar":
+                                                                if(data.points[0].data.orientation=="h")
+                                                                    idx=data.points[0].data.y.indexOf(data.points[0].y.toString());
+                                                                else
+                                                                    idx=data.points[0].data.x.indexOf(data.points[0].x.toString());
+                                                                break;
+                                                        }
+                                                        if( data.points[0].data.graphic_link_pattern[idx]!=false && data.points[0].data.graphic_link_pattern[idx]!=undefined){
+                                                            var url=data.points[0].data.graphic_link_pattern[idx];
```

Comment:
Seems like the local variable `graphic_link_pattern` (not the variable read from the configuration document) is not a single "link pattern" but instead a set of urls. Let's rename this variable to represent that better, how about `graphic_urls` (or `graphic_clickable_links` which is consistent with `legend_clickable_links` below)?

Reply:
I will rename it to graphic_clickable_links to keep it consistent with legend_clickable_links